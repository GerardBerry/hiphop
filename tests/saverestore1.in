!eval-begin;
var state = example.prg.save();
!eval-end;
;
!eval-begin;
example.prg.restore(state);
!eval-end;
;
!eval-begin;
example.prg.restore(state);
!eval-end;
;
!eval-begin;
example.prg.restore(state);
!eval-end;
;