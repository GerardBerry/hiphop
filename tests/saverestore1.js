"use hopscript"

const machine = require("./p18.js").prg;
let state = machine.save();

console.log(machine.react());

machine.restore(state);
console.log(machine.react());

machine.restore(state);
console.log(machine.react());

machine.restore(state);
console.log(machine.react());
