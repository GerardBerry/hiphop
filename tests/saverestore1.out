P18> ;
--- Output: not_S1_and_not_S2
P18> ;
--- Output: not_S1_and_not_S2
P18> ;
--- Output: not_S1_and_not_S2
P18> ;
--- Output: not_S1_and_not_S2
